simple-verifier/src/main/java/com/example/simple_verifier/MainActivity.kt

```diff
@@ -0,0 +1,243 @@
+            ContextCompat.checkSelfPermission(
+                this.applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        }
+
+        if (permissionsNeeded.isNotEmpty()) {
+            permissionsLauncher.launch(
+                permissionsNeeded.toTypedArray()
+            )
+        }
+    }
+
+    private fun isAllPermissionsGranted(): Boolean {
+        // If any permission is not granted return false
+        return appPermissions.none { permission ->
+            ContextCompat.checkSelfPermission(
+                this.applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        }
+    }
+}
+
+@Composable
+fun LibraryApp(
+    modifier: Modifier = Modifier,
+    supportFragmentManager: FragmentManager,
+    enableReaderPermissions: () -> Unit
+) {
+    Surface(modifier) {
+        LibraryToolboxScreen(modifier = modifier, supportFragmentManager = supportFragmentManager, enableReaderPermissions = enableReaderPermissions)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun LibraryToolboxScreen(
+    modifier: Modifier = Modifier,
+    supportFragmentManager: FragmentManager,
+    enableReaderPermissions: () -> Unit
+) {
+    Scaffold(
+        topBar = {
+            CenterAlignedTopAppBar(
+                colors = largeTopAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.surfaceContainer,
+                    titleContentColor = MaterialTheme.colorScheme.primary,
+                ),
+                title = {
+                    Text(
+                        style = MaterialTheme.typography.headlineMedium,
+                        text = "Age Assurance Example",
+                        color = MaterialTheme.colorScheme.onSurface
+                    )
+                }
+            )
+        },
+        containerColor = MaterialTheme.colorScheme.surface
+    ) { innerPadding ->
+
+        Toolbox(
+            on21Clicked = {
+                enableReaderPermissions()
+                val mdocReaderPrompt = MdocReaderPrompt(MdocReaderSettings.Builder()
+                    .setAgeVerificationType(AgeVerificationType.Over21)
+                    .build())
+                mdocReaderPrompt.show(supportFragmentManager, null)
+            },
+            on18Clicked = {
+                enableReaderPermissions()
+                val mdocReaderPrompt = MdocReaderPrompt(MdocReaderSettings.Builder()
+                    .setAgeVerificationType(AgeVerificationType.Over18)
+                    .build())
+                mdocReaderPrompt.show(supportFragmentManager, null)
+            },
+            modifier = Modifier
+                .padding(innerPadding)
+                .padding(vertical = 20.dp)
+                .padding(horizontal = 20.dp)
+                .fillMaxWidth()
+        )
+    }
+}
+
+@Composable
+private fun Toolbox(
+    on21Clicked: () -> Unit,
```

Comment:
Nit: I had to do a double-take when I saw parameter name, maybe spell it out so it's `onOver21ButtonClicked` instead.